vortex-layout/src/layouts/flat/mod.rs

```diff
@@ -186,6 +186,65 @@ impl FlatLayout {
     }
 }
 
+/// Display wrapper for the array tree metadata stored in a FlatLayout.
+pub struct ArrayTreeDisplay {
+    array_tree: Option<ByteBuffer>,
+    segment_id: SegmentId,
```

Comment:
We only ever use the segment id if `array_tree` is not set. Should these be diff cases represented by an enum?